### PR TITLE
Add ifdefs for is_reinitialized in FEEval

### PR DESCRIPTION
**Pull request by peterrum**
referencing https://github.com/dealii/dealii/pull/13258#discussion_r789142926

### CODE REVIEW

**Comment by drwells** [comment]
This works but I would really prefer that we fix it another way. Could we just set `is_reinitialized` to be `true` whenever we are not in debug mode?

**Comment by peterrum** [comment]
This is how we do it in many other places in FEEval, e.g., https://github.com/dealii/dealii/blob/0804715bf1cd446a383fc86128c979fc8c05ace1/include/deal.II/matrix_free/fe_evaluation.h#L4485-L4488

**Review by drwells — APPROVED** [review]
This change is consistent with the way we do similar workarounds so lets go ahead and merge. We can improve the logic later.

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -4987,7 +4987,9 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
   submit_value(const Tensor<1, n_components_, VectorizedArrayType> val_in,
                const unsigned int                                  q_point)
 {
+#  ifdef DEBUG
   Assert(this->is_reinitialized, ExcNotInitialized());
+#  endif
```
[the remark this answers is not in the document]
Comment (kronbichler) [reply include/deal.II/matrix_free/fe_evaluation.h:4992]: The "problem" is that we redefine it for the test suite, https://github.com/dealii/dealii/blob/5478749bc821598a9486cf53abce53a14da33a35/tests/tests.h#L72-L73

**Review by kronbichler — APPROVED** [review]

**Pull request merged**